lua/gitlinker/git.lua

```diff
@@ -238,6 +238,31 @@ function M.get_closest_remote_compatible_rev(remote)
   return nil
 end
 
+local function get_remote_branch(remote)
+  local errs = {
+    string.format("Failed to retrieve remote branch for remote '%s'", remote),
+  }
+  local symRef
+  local p = job:new({
+    command = "git",
+    args = { "symbolic-ref", "-q", "HEAD" },
+  })
+  p:after_success(function(j)
+    symRef = j:result()[1]
+  end)
+  p:sync()
+  local trackingBranch
+  local p = job:new({
+    command = 'git',
+    args = { 'for-each-ref', '--format=%(upstream:short)', symRef },
+  })
+  p:after_success(function(j)
+    trackingBranch = (j:result()[1]):match(remote .. '/(.+)$')
+  end)
+  p:sync()
+  return trackingBranch or 'origin/master'
```

Comment:
don't assume anything, if we can't find the remote branch might as well error out
```suggestion
  return trackingBranch
```